### PR TITLE
Testem v1.0.0 & detect global errors

### PR DESCRIPTION
**Pull request by johanneswuerbach**
Testem can now instrument the window, before the test framework is
loaded and manually hooked into the framework later. This allows
catching syntax and other global errors.

Fixes: https://github.com/ember-cli/ember-cli/issues/5175

The current master includes some bigger changes and requires some testing time. I would love to merge this after the next stable is shipped. Do you have any rough estimate when the next ember-cli stable might be cut?

### CODE REVIEW

**Comment by stefanpenner** [comment]
@nathanhammond because this is related to tests, I suspect this doesn't interfere with your work?

**Comment by stefanpenner** [comment]
@johanneswuerbach can you also make sure testem.js is loaded (in tests/index.html) first.

**Comment by nathanhammond** [comment]
This is lower on my priority list. Eventually I want to make all of tests work the same way, but that's tertiary (app, then vendor, then tests).

File: package.json [hunk 1]
```diff
@@ -111,7 +111,7 @@
     "silent-error": "^1.0.0",
     "symlink-or-copy": "^1.0.1",
     "temp": "0.8.3",
-    "testem": "0.9.11",
+    "testem": "testem/testem",
```
Comment (nathanhammond) [inline package.json]: Moving to master will make it difficult for repeatable builds. Not a fan of this.

Comment (stefanpenner) [inline package.json]: ya, we wont release that. (will break with firewalled peeps)

Comment (johanneswuerbach) [inline package.json]: Yes, I'm working on testem v1. The git ref is just here until the release is cut.

Comment (stefanpenner) [inline package.json]: :+1:

**Comment by johanneswuerbach** [comment]
https://github.com/ember-cli/ember-cli/pull/5254#issuecomment-166761950 done.

**Comment by stefanpenner** [comment]
> @stefanpenner #5254 (comment) done.

:+1:

**Comment by johanneswuerbach** [comment]
Updated with a published release candidate https://github.com/testem/testem/releases/tag/v1.0.0-rc.1

> The current testem master includes some bigger changes and requires extended testing time. I would love to merge this after the next stable is shipped. Do you have any rough estimate when the next ember-cli stable might be cut?

Any update on this?

**Comment by rwjblue** [comment]
We will likely be cutting another beta release soon, but I do not expect to ship a stable release in the next week or two at least.

**Comment by johanneswuerbach** [comment]
Two weeks testings would work for me. Your call whether to include this PR or wait for testem stable, I would :heart: when it would be merged as ember-cli seems to be one of the biggest consumers today and having more hands testing the 1.0 would be better.

**Comment by rwjblue** [comment]
I'm happy to land this now, and release in the next beta (should get feedback pretty quickly).

@stefanpenner - You happy with that plan?

**Comment by stefanpenner** [comment]
I'm fine with shipping this as part of the next stable stable-1-13 release and merging into master.

testem master seems to be reasonably stable, and the faster we catch stuff the better.

**Comment by johanneswuerbach** [comment]
Ready from my side. I'll publish once we had ~a week of testing and everything feels well.

**Comment by rwjblue** [comment]
Thanks @johanneswuerbach! I'm going to try to ship another beta soon, and then we can release a stable version after a couple days.

**Pull request merged**